contracts/services/reverse-splitter/src/msg.rs

```diff
@@ -0,0 +1,424 @@
+        return Err(ServiceError::ConfigurationError(
+            "No split configuration provided.".to_string(),
+        ));
+    }
+
+    let mut denom_set = HashSet::new();
+    for split in splits {
+        split.account.to_addr(api)?;
+        // Note: can't validate denom without the deps
+
+        // Ensure splits are unique in split configs
+        let key = format!("{:?}|{:?}", split.denom, split.account);
+        if !denom_set.insert(key) {
+            return Err(ServiceError::ConfigurationError(format!(
+                "Duplicate split '{:?}|{:?}' in split config.",
+                split.denom, split.account
+            )));
+        }
+
+        match &split.amount {
+            UncheckedSplitAmount::FixedAmount(amount) => {
+                if amount == Uint128::zero() {
+                    return Err(ServiceError::ConfigurationError(
+                        "Invalid split config: amount cannot be zero.".to_string(),
+                    ));
+                }
+
+                if split.factor.is_some() {
+                    return Err(ServiceError::ConfigurationError(
+                        "Invalid split config: a factor cannot be specified with an amount."
+                            .to_string(),
+                    ));
+                }
+            }
+            UncheckedSplitAmount::FixedRatio(ratio) => {
+                if ratio == Decimal::zero() {
+                    return Err(ServiceError::ConfigurationError(
+                        "Invalid split config: ratio cannot be zero.".to_string(),
+                    ));
+                }
+                if split.denom == *base_denom && ratio != Decimal::one() {
+                    return Err(ServiceError::ConfigurationError(
+                        "Invalid split config: fixed ratio for base denom must be 1.".to_string(),
+                    ));
+                }
+            }
+            UncheckedSplitAmount::DynamicRatio { contract_addr, .. } => {
+                api.addr_validate(contract_addr)?;
+                if split.denom == *base_denom {
+                    return Err(ServiceError::ConfigurationError(
+                        "Invalid split config: ratio for base denom cannot be a dynamic one."
+                            .to_string(),
+                    ));
+                }
+            }
+        }
+
+        if let Some(factor) = split.factor {
+            if factor == 0 {
+                return Err(ServiceError::ConfigurationError(
+                    "Invalid split config: factor cannot be zero.".to_string(),
+                ));
+            }
+        }
+    }
+
+    // If there are ratios, we only allow an amount to be set for the base denom
+    if splits
+        .iter()
+        .any(|s| !matches!(s.amount, UncheckedSplitAmount::FixedAmount(..)))
+        && splits.iter().any(|s| {
+            matches!(s.amount, UncheckedSplitAmount::FixedAmount(..)) && s.denom != *base_denom
+        })
```

Comment:
We already doing a loop above, we can have a flag set to true at the above loop if `any !UncheckedSplitAmount::FixedAmount(..)`, instead of doing another loop here.

Reply:
fixed